### DISCUSSION TITLE
Using a query key lookup table instead of encryptions...

### DISCUSSION THREAD

**Discussion by Rolands-Laucis** [Rolands-Laucis/Socio#7] [Ideas]
So this is my classic case of not seeing the wood for the trees.

Someone recently pointed out to me that Socio could hash the SQL query strings instead of encrypting them, such that they only operate as keys for a lookup table like a hashmap on the backend. That way its just an arbitrary key, that can even be just a counter.

This would make the query strings small, reducing the JS bundle size and network traffic. In addition, remove devs bad gut feeling of SQL sitting on the front-end. It also happens to make the server more performant and the code smaller.

I've since created another branch experimenting with such an idea. However, there is a big **problem -** the Query Map/Lookup table itself needs to exist somewhere, somehow on the backend. 

Since building the website is a different process than running it, and the build process creates the Map, but only the run process actually needs it, it needs to somehow persist between these processes. The only way i've been able to come up with is - the File System. Generate a json or whatever of the Map and read it at the start of running the website. However, imo this is problematic too. If I were using Socio as a dev, it would be unnerving to see it spawn files in my project. Additionally, this file should be untracked by version control systems, e.g. git, so that requires forethought and config.

I could create the file in the build folder, but then what about the dev server? Idk how Vite's dev server reloading would work - would it rerun the socio plugin and reread the new Map file? Also your backend server is going to need to know exactly where this file is in both Dev and Prod env. which complicates things. And this file would be static, since it would only be read at the start of the webserver.

Perhaps this is a small price to pay for the benefits listed in the 3rd paragraph. What do you think?

**Comment by KaruroChori** [comment]
This change would affect one of the maybe unintentional main benefits of the current approach: currently we can have multiple independent yet trusted (and/or subject to permissions and auth methods) clients working with the same back-end, as long as the db schema is not affected.
